### PR TITLE
Release/v0.4.0

### PR DESCRIPTION
**Pull request by gumil**
## Changes
- `SavedValueHolder.savedValues` now returns `Map`
- `initialState` visibility changed to a public field
- Refactor `process` to `dispatch` #50 
- Add support to Coroutines `StateFlow`
- Fix deprecations

### Dependency Updates
- Artifacts
  - Kotlin Coroutines v1.3.7 

- Tooling
  - AGP v4.0.0
  - Node plugin v2.2.4
  - Detekt v1.10.0-RC1
  - JaCoCo v0.8.5
  - Android Build Tools v30.0.0

- Sample Apps
  - Retrofit v2.9.0
  - minSdkVersion 23
  - RxBindin v3.1.0
  - Material Components v1.3.0-alpha01
  - Navigation v2.3.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,5 +118,10 @@ Some of the topics covered are:
 
 ### Sample projects
 * [Android App](sample/app) - Android use cases
+    - [auth](sample/app/src/main/kotlin/dev/gumil/kaskade/sample/auth) - Rx use case
```
Comment (gumil) [inline README.md:121]: Codacy found an issue: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

**Comment by gumil** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 13
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048&bid=18867197)

File: README.md [hunk 2]
```diff
@@ -118,5 +118,10 @@ Some of the topics covered are:
...
+    - [network](sample/app/src/main/kotlin/dev/gumil/kaskade/sample/network) - Coroutines and LiveData use case
+    - [todo](sample/app/src/main/kotlin/dev/gumil/kaskade/sample/todo) - Vanilla kaskade use case
 * [Kotlin Console](sample/console) - Kotlin only project
-* [Giphy Android Use Case](https://github.com/gumil/giphy) - Android master-detail flow use case
+* [Giphy Android Use Case](https://github.com/gumil/giphy)
+    - Android master-detail 
+    - Reactive FlowBinding and StateFlow use case
```
Comment (gumil) [inline README.md:127]: Codacy found an issue: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

Comment (gumil) [inline README.md:127]: Codacy found an issue: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

File: README.md [hunk 3]
```diff
@@ -118,5 +118,10 @@ Some of the topics covered are:
 
 ### Sample projects
 * [Android App](sample/app) - Android use cases
+    - [auth](sample/app/src/main/kotlin/dev/gumil/kaskade/sample/auth) - Rx use case
+    - [network](sample/app/src/main/kotlin/dev/gumil/kaskade/sample/network) - Coroutines and LiveData use case
+    - [todo](sample/app/src/main/kotlin/dev/gumil/kaskade/sample/todo) - Vanilla kaskade use case
```
Comment (gumil) [inline README.md:123]: Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

File: README.md [hunk 4]
```diff
@@ -118,5 +118,10 @@ Some of the topics covered are:
 
 ### Sample projects
 * [Android App](sample/app) - Android use cases
+    - [auth](sample/app/src/main/kotlin/dev/gumil/kaskade/sample/auth) - Rx use case
+    - [network](sample/app/src/main/kotlin/dev/gumil/kaskade/sample/network) - Coroutines and LiveData use case
```
Comment (gumil) [inline README.md:122]: Codacy found an issue: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

File: README.md (hunk 3 above)
Comment (gumil) [inline README.md:123]: Codacy found an issue: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

File: kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package dev.gumil.kaskade.coroutines
+
+import kotlinx.coroutines.CoroutineScope
+
+@Suppress("UnusedPrivateMember")
```
Comment (gumil) [inline kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt:5]: Codacy found an issue: [UndocumentedPublicFunction - [runTest] at /src/kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt:5:1](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

File: README.md (hunk 3 above)
Comment (gumil) [inline README.md:123]: Codacy found an issue: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

File: kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package dev.gumil.kaskade.coroutines
```
Comment (gumil) [inline kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt:1]: Codacy found an issue: [FinalNewline - [] at /src/kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt:1:1](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

File: README.md (hunk 4 above)
Comment (gumil) [inline README.md:122]: Codacy found an issue: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

File: README.md (hunk 1 above)
Comment (gumil) [inline README.md:121]: Codacy found an issue: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

File: README.md [hunk 7]
```diff
@@ -118,5 +118,10 @@ Some of the topics covered are:
...
+    - [todo](sample/app/src/main/kotlin/dev/gumil/kaskade/sample/todo) - Vanilla kaskade use case
 * [Kotlin Console](sample/console) - Kotlin only project
-* [Giphy Android Use Case](https://github.com/gumil/giphy) - Android master-detail flow use case
+* [Giphy Android Use Case](https://github.com/gumil/giphy)
+    - Android master-detail
```
Comment (gumil) [inline README.md:126]: Codacy found an issue: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

Comment (gumil) [inline README.md:126]: Codacy found an issue: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

**Comment by gumil** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048&bid=18867197)

File: kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt (hunk 6 above)
Comment (gumil) [inline kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt:1]: Codacy found an issue: [FinalNewline - [] at /src/kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt:1:1](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

File: kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt (hunk 5 above)
Comment (gumil) [inline kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt:5]: Codacy found an issue: [UndocumentedPublicFunction - [runTest] at /src/kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt:5:1](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

File: README.md [hunk 8]
```diff
@@ -118,5 +118,10 @@ Some of the topics covered are:
 
 ### Sample projects
 * [Android App](sample/app) - Android use cases
+  * [auth](sample/app/src/main/kotlin/dev/gumil/kaskade/sample/auth) - Rx use case
+  * [network](sample/app/src/main/kotlin/dev/gumil/kaskade/sample/network) - Coroutines and LiveData use case
+  * [todo](sample/app/src/main/kotlin/dev/gumil/kaskade/sample/todo) - Vanilla kaskade use case
```
Comment (gumil) [inline README.md:123]: Codacy found an issue: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/gumil/Kaskade/pullRequest?prid=5764048)

File: kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt (hunk 5 above)
Comment (jasvilladarez) [inline kaskade/coroutines/src/iosArm64Test/kotlin/dev/gumil/kaskade/coroutines/runTest.kt:5]: Add comment that this is for detekt

**Review by jasvilladarez — APPROVED** [review]

**Pull request merged**